### DISCUSSION TITLE
Best practice to set options base on nested screen

### DISCUSSION THREAD

**Discussion by eidan66** [react-navigation/react-navigation#11820] [Q&A] [answered]
Hey!
I have this stack:
```txt
- DrawerStack
  -  BottomTabs
    - HomeStack
      - HomeScreen
    - SettingStack
      - SettingScreen
```

At the `DrawerStack` i have a header with:
* headerLeft (the drawer icon)
* Title

I want to set the `headerRight` option (at `DrawerStack` header) based on the nested screen (e.g. `HomeScreen`).

What do you think the best practice is to do it?
If i will do it inside the `HomeScreen` component, I'll call to `navigation.getParent().getParent()` and I don't think it is a good practice.

React Navigation V6.

**Comment by satya164** [comment, marked as answer]
Use `getParent` with an ID, e.g. `getParent('MyNavigator')`

https://reactnavigation.org/docs/navigation-prop/#getparent

**Reply by eidan66** [reply]
Thanks!
Is there an option to set all those configs (adding the right header) on the `DrawerStack` based on the nested child (`HomeScreen` for example)?

I saw now that I'm duplicating the code too much time
